### ISSUE TITLE
Circuit Relay test is failing.

### ISSUE THREAD

**Issue by diasdavid** [libp2p/js-libp2p#198]
@jacobheun could this have been introduced with the refactor? It seems that the muxer on the connection is not being correctly set up when a Relay connection happens.

```
  1) circuit relay
       "after all" hook:
     Uncaught TypeError: Cannot read property 'muxer' of undefined
      at each (node_modules/libp2p-switch/src/index.js:113:14)

```

//cc @dryajov

**Comment by jacobheun** [comment]
@diasdavid I'll take a look at this, I think it's just an issue with resiliency when a peer hangs up while we're also shutting down. If a peer hangs up at the same time we're shutting down (such as with test cases), it's possible we could remove the muxed connection on the hang up, as we're iterating on them to shut each of them down, making the connection undefined.  I'll make sure this is the case and improve the resiliency to prevent this error.

**Comment by jacobheun** [comment]
My assumption was correct, I have the update ready here https://github.com/libp2p/js-libp2p-switch/pull/260.  The PR also links to the passing CI tests for libp2p via the chore/update-switch branch.

**Comment by jacobheun** [comment]
@diasdavid this is fixed in the latest release of switch, 0.40.3.  There's another issue with the tests failing. I'll see if I can figure out what's going on there.

```
  1) transports TCP + WebSockets + WebRTCStar nodeAll.dial nodeWStar using PeerInfo:
     Error: Timeout of 40000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.
  2) transports TCP + WebSockets + WebRTCStar nodeAll.hangUp nodeWStar using PeerInfo:
      Uncaught AssertionError: expected [] to have a length of 1 but got 0
      + expected - actual
      -0
      +1
  
      at forEach (test/transports.node.js:455:61)
      at Array.forEach (<anonymous>)
      at check (test/transports.node.js:454:28)
      at Timeout.setTimeout [as _onTimeout] (test/transports.node.js:506:26)
```

https://ci.ipfs.team/blue/organizations/jenkins/libp2p%2Fjs-libp2p/detail/chore%2Fupdate-switch/5/pipeline

**Comment by jacobheun** [comment]
js-libp2p-webrtc-star v0.15.2 is breaking the tests, looking into it.

**Comment by diasdavid** [comment]
Thank you @jacobheun :)

**Comment by jacobheun** [comment]
The underlying issue appears to be with multistream-select and block-streams, which were added via https://github.com/libp2p/js-libp2p-webrtc-star/pull/145, they are not playing well together. 

I am looking into why that's the case. I have tested this against `pull-block` and `block-streams` and both consistently hang, which explains the hang in the tests we are seeing in libp2p.

@diasdavid I think we might want to rollback the update to libp2p-webrtc-star until this is resolved as the multistream connections won't work. I know @ya7ya is also looking into updates for block streams to fix https://github.com/ipfs/js-ipfs/issues/1379, so it might be good to roll those together. If I can get the multistream-select/block-stream fix done, it should make it so we can use pull-block (I am adding tests to multistream-select using pull-block).

### additional info
I am seeing a clear, consistent, order of operations difference in the calls between block and non block usage.

**without block streams (working)**
```
  mss:dialer     (1plvt5) dialer handle conn +0ms
  mss:dialer     (1plvt5) writing multicodec: /multistream/1.0.0 +1ms
  mss:listener   (dwutnx) listener handle conn +0ms
  mss:listener   (dwutnx) writing multicodec: /multistream/1.0.0 +0ms
  mss:listener   (dwutnx) received ack: /multistream/1.0.0 +3ms
  mss:listener   (dwutnx) adding handler: /monkey/1.0.0 +2ms
  mss:dialer     (1plvt5) received ack: /multistream/1.0.0 +209ms
  mss:dialer     (1plvt5) handshake success +1ms
  mss:dialer     (1plvt5) dialer select /monkey/1.0.0 +0ms
  mss:dialer     (1plvt5) writing multicodec: /monkey/1.0.0 +0ms
# truncated here, but the logs continue
```

**with block streams (hanging)**
```
  mss:dialer     (1sx1xq) dialer handle conn +0ms
  mss:dialer     (1sx1xq) writing multicodec: /multistream/1.0.0 +0ms
  mss:listener   (1xawf5) listener handle conn +0ms
  mss:listener   (1xawf5) writing multicodec: /multistream/1.0.0 +0ms
  mss:dialer     (1sx1xq) received ack: /multistream/1.0.0 +202ms
  mss:dialer     (1sx1xq) handshake success +0ms
  mss:dialer     (1sx1xq) dialer select /monkey/1.0.0 +0ms
  mss:dialer     (1sx1xq) writing multicodec: /monkey/1.0.0 +0ms
  mss:listener   (1xawf5) received ack: /multistream/1.0.0 +3ms
  mss:listener   (1xawf5) adding handler: /monkey/1.0.0 +1ms
# hangs here
```

**Comment by diasdavid** [comment]
@jacobheun thank you so much for the debugging. I've added you and @pgte to have publish access of libp2p-webrtc-star so that you can unpublish the right version (https://www.npmjs.com/package/libp2p-webrtc-star/access) I'm afraid that with limited time and massive context switch that I might do a poor job of reverting those changes.

**Comment by jacobheun** [comment]
libp2p-webrtc-star 0.15.1 has been marked as latest and the 0.15.2 changes have been rolled back into master. This issue will be resolved with the release of #203
